### PR TITLE
Use write-ahead log for RepositoryContent.

### PR DESCRIPTION
**Pull request by timbru**
[draft] reviewing allowed, but can wait for the following:

Will deploy to benchmark environment to test upgrade and effect.
Still need to update the developer docs.

### CODE REVIEW

**Comment by timbru** [comment]
Deployed to benchmarking VM with 5000 CAs with 10 ROAs each.

The system is responsive and CPU load is low, except for when publication happens. With the additional log messages added we can see that this is due to updating the RRDP state and writing new RRDP files for every update.

This should be solved when https://github.com/NLnetLabs/krill/issues/693 is implemented. That will allow us to defer RRDP updates and include multiple publisher deltas in one. Furthermore, this will mean that the RRDP are not blocking publisher.

The RRDP update process itself can also be optimised more. Currently it's creating the snapshot XML twice (and for a 280 MB XML file this is what's blocking the system). Once to get the hash to include in the notification, and once to write the actual file. We may want to optimise this, and perhaps use the RRDP support from rpki-rs rather than the older RRDP XML support in krill itself.

**Comment by timbru** [comment]
This needs more work - it's currently saving the entire new snapshot in an RrdpUpdate change. This naive implementation ends up using way too much data. So, will change this and only then mark this for review. The CPU usage seems quite improved at least.

**Comment by timbru** [comment]
The deltas are now much smaller. Overall performance is still not good enough because creating new RRDP snapshots is still too expensive and happens too often. This will be fixed in a separate PR for issue #693.

This should be good to merge now, unless other issues are raised during review of course.

File: src/commons/error.rs
Comment (ximon18) [inline src/commons/error.rs:662]: You distinguish wal store error from aggregate store error here, but above you use the same description string for both. This seems inconsistent.

File: src/commons/eventsourcing/wal.rs
Comment (ximon18) [inline src/commons/eventsourcing/wal.rs:139]: Minor: This is doing something (i.e. warming up) rather than being "warm" so wouldn't it be better named `warmup()` or similar?

**Review by ximon18 — APPROVED** [review]

File: src/commons/error.rs
Comment (timbru) [reply src/commons/error.rs:662]: updated the error text to be a bit more clear (none of this is likely to be seen by users, but still)

File: src/commons/eventsourcing/wal.rs
Comment (timbru) [reply src/commons/eventsourcing/wal.rs:139]: I prefer to keep it like this. This is consistent with `AggregateStore::warm`. Fwiw.. "warm" was intended as a verb, and the imperative at that. For being warm I would use something like "is_warm" or "warmed" (i.e. past tense, not the imperative).

File: src/commons/eventsourcing/wal.rs
Comment (timbru) [reply src/commons/eventsourcing/wal.rs]: fixed

File: src/commons/eventsourcing/wal.rs
Comment (timbru) [reply src/commons/eventsourcing/wal.rs:139]: To be clear - I think `warmup` would have been better, but I don't think that `warm` on its own is too problematic. I am happy to change after all the stacked PRs are merged - because I fear conflicts.

**Pull request merged**
